plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -442,6 +442,16 @@ def _onSlicingFinishedMessage(self, message):
         self.backendStateChange.emit(BackendState.Done)
         self.processingProgress.emit(1.0)
 
+        for line in self._scene.gcode_list:
+            replaced = line.replace("{print_time}", str(Application.getInstance().getPrintInformation().currentPrintTime.getDisplayString(DurationFormat.Format.ISO8601)))
+            replaced = replaced.replace("{filament_amount}", str(Application.getInstance().getPrintInformation().materialLengths))
+            replaced = replaced.replace("{filament_weight}", str(Application.getInstance().getPrintInformation().materialWeights))
+            # TODO: calculate filament cost
+            replaced = replaced.replace("{filament_cost}", "Not yet implemented")
+            replaced = replaced.replace("{jobname}", str(Application.getInstance().getPrintInformation().jobName))
+
+            self._scene.gcode_list[self._scene.gcode_list.index(line)] = replaced
+
```

Comment:
There is a better place for all this: `plugins.CuraEngineBackend.StartSliceJob._buildGlobalSettingsMessage`.

In there, we already create a few keywords such as `{time}`, `{date}` and `{day}`. You could add your new keywords to it very easily. What's more, that function performs the replacing only on the actual start and end g-codes rather than on all of the g-code, and does it in one query rather than a query for each line, which is much more efficient.